geps/gep-713.md

```diff
@@ -560,50 +560,536 @@ ties:
+
+There is a pause while Ana's mind races through increasingly unpleasant
+possibilities.
+
+**Ana**: _I don't even know where to start here. How long did that
+RetryPolicy go in? Is it the only thing like it?_
+
+**Charlie**: _Didn’t look closely before deleting it, but I think it said a few
+months ago. And there are lots of different kinds of policy and lots of
+individual policies, hang on a minute..._
+
+**Charlie**: _Looks like about 47 for your chunk of the world, a couple hundred
+system-wide._
+
+**Ana**: 😱 _Can you tell me what they’re doing for each of our services? I
+can’t even_ look _at these things._ 😕
+
+**Charlie**: _That's gonna take awhile. Our tooling to show us which policies
+bind to a given workload doesn't go the other direction._
+
+**Ana**: _...wait. You have to_ build tools _to know if retries are turned on??_
+
+Pause.
+
+**Charlie**: _Policy attachment is more complex than we’d like, yeah._ 😐
+_Look, how ‘bout roll back your `baker` change for now? We can get together in
+the morning and start sorting this out._
+
+Ana shakes her head and rolls back her edits to the `baker` Deployment, then
+sits looking out over the lake as the deployment progresses.
+
+**Ana**: _Done. Are things happier now?_
+
+**Charlie**: _Looks like, thanks. Reckon you can get back to your sailboard._ 🙂
+
+Ana sighs.
+
+**Ana**: _Wish I could. Wind’s died down, though, and it'll be dark soon.
+Just gonna head home._
+
+**Charlie**: _Ouch. Sorry to hear that._ 😐
+
+One more look out at the lake.
+
+**Ana**: _Thanks for the help. Wish we’d found better answers._ 😢
+
+### The Problem, restated
+What this parable makes clear is that, in the absence of information about what
+Policy is affecting an object, it’s very easy to make poor decisions.
+
+It’s critical that this proposal solve the problem of showing up to three things,
+listed in increasing order of desirability:
+- _That_ some Policy is affecting a particular object
+- _Which_ Policy is (or Policies are) affecting a particular object
+- _What_ settings in the Policy are affecting the object.
+
+In the parable, if Ana and Charlie had known that there were Policies affecting
+the relevant object, then they could have gone looking for the relevant Policies
+and things would have played out differently. If they knew which Policies, they
+would need to look less hard, and if they knew what the settings being applied
+were, then the parable would have been able to be very short indeed.
+
+(There’s also another use case to consider, in that Charlie should have been able
+to see that the Policy on the namespace was in use in many places before deleting
+it.)
+
+To put this another way, Policy Attachment is effectively adding a fourth Persona,
+the Policy Admin, to Gateway API’s persona list, and without a solution to the
+discoverability problem, their actions are largely invisible to the Application
+Developer. Not only that, but their concerns cut across the previously established
+levels.
```

Comment:
I'm wondering if this problem would be less severe if there were certain restrictions on where and how policy attachments are used?

Seems like one thing that should always be possible is that the owner of a resource should be able to override any inherited policy attachments. Does it really make sense that a policy attachment can override an explicit setting in a resource? What is the use case for having explicit settings in the user's resource that get ignored and changed under the covers because of some policy attachment somewhere?

I'm not sure what properties of a resource the spec says can be controlled by inherited policy attachments, but I would say there "should be" only two things:
1. default values of fields in the resource (first class API fields)
2. fields that can be set with direct policy attachments (these are fields that presumably have been added to support things that are not directly supported in the resources API itself)

If that was the case, then the owner of a resource would always be able to explicitly override any inherited policy attachments that don't work, either in the resource itself or in a direct policy attachment sitting next to the resource.

Reply:
Being able to override specific settings can be _very_ useful in clusters with strict rules about things, although I agree that it's probably better in those cases to reject the create or change that would have an undesirable setting using a Validating Admission Controller - faster feedback is _much_ better than silently changing a value.

But, this override need comes from direct experience @robscott and I have had with previous iterations of similar ideas, sysadmins (in Gateway API's personas, Cluster Admins or even Infra Admins) sometimes want to override settings so that users can't take certain actions, and this is a mechanism to allow that to happen.